cmd/cluster-agent/subcommands/start/command.go

```diff
@@ -477,3 +493,26 @@ func initializeRemoteConfig(ctx context.Context) (*rcclient.Client, *rcservice.S
 
 	return rcClient, rcService, nil
 }
+
+func registerChecks() {
+	// Required checks
+	registerCheck(true, cpu.CheckName, cpu.New)
+	registerCheck(true, memory.CheckName, memory.New)
+	registerCheck(true, uptime.CheckName, uptime.New)
+	registerCheck(true, io.CheckName, io.New)
+	registerCheck(true, filehandles.CheckName, filehandles.New)
+
+	// Flavor specific checks
+	registerCheck(kubernetesapiserver.Enabled, kubernetesapiserver.CheckName, kubernetesapiserver.New)
+	registerCheck(ksm.Enabled, ksm.CheckName, ksm.New)
+	registerCheck(helm.Enabled, helm.CheckName, helm.New)
+	registerCheck(disk.Enabled, disk.CheckName, disk.New)
+	registerCheck(orchestrator.Enabled, orchestrator.CheckName, orchestrator.New)
+	registerCheck(winproc.Enabled, winproc.CheckName, winproc.New)
+}
+
+func registerCheck(enabled bool, name string, factory func() check.Check) {
```

Comment:
Since this is repeated twice, would it make sense to move it into the `corecheckLoader` package? (as `RegisterCheckIf` perhaps)